### PR TITLE
Release/1.35.0

### PR DESCRIPTION
**Pull request by mike-kelly-methods**
### What

Release includes:

- `JSON web key set endpoint` to return `JWT Key ID` and associated `RSA Public Signing Key (`AWS Cognito User Pool` specific)`
- Renaming of `Group` entity response fields
- Masking of internal server errors

### How to review

Ensure all tests pass.

### Who can review

!me

### CODE REVIEW

File: api/jwks_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		},
+	}
+
+	r := httptest.NewRequest(http.MethodGet, jwksEndpoint, nil)
+
+	resp, err := api.CognitoPoolJWKSHandler(context.Background(), w, r)
+
+	Convey("Request json web key set - 500 response", t, func() {
+		e := err.Errors[0].Error()
+		println(e)
```
Comment (daiLlew) [inline api/jwks_test.go]: Can we remove this please?

File: features/steps/identity_component.go [hunk 2]
```diff
@@ -29,6 +31,7 @@ type IdentityComponent struct {
 	apiFeature              *componenttest.APIFeature
 	CognitoClient           *cognitoMock.CognitoIdentityProviderClientStub
 	AuthorisationMiddleware authorisation.Middleware
+	JWKSHandler				*jwksMock.JWKSIntMock
```
Comment (daiLlew) [inline features/steps/identity_component.go]: Formatting.

File: jwks/jwks.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+type JWKS struct {
+	Keys []JsonKey `json:"keys"`
+}
+
+// JWKSGetKeyset primary package method which retrives the json web key set for cognito user pool
+func (j JWKS) JWKSGetKeyset(awsRegion, poolId string) (*JWKS, error) {
+	resp, err := http.Get(fmt.Sprintf(jwksURL, awsRegion, poolId))
+	if err != nil {
+		return nil, err
+	}
```
Comment (daiLlew) [inline jwks/jwks.go:51]: We need to add a `defer resp.Body.close()` here as there is a potential resource leak.

https://pkg.go.dev/net/http#Response

File: jwks/jwks.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+// JWKToRSAPublicKey transforms key data to PKIX, ASN.1 DER form
+func (j JWKS) JWKToRSAPublicKey(jwk JsonKey) (string, error) {
+	if jwk.Kty != RSAAlgorithm {
+		return "", errors.New(models.JWKSUnsupportedKeyTypeDescription)
+	}
+
+	// decode the base64 bytes for n
+	nb, err := base64.RawURLEncoding.DecodeString(jwk.N)
+	if err != nil {
+		log.Println(err)
```
Comment (daiLlew) [inline jwks/jwks.go]: Remove this println

File: jwks/jwks.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+	// Use default exponent -> 65537
+	if jwk.E == RSAExponentAQAB || jwk.E == RSAExponentAAEAAQ {
+		der, err := x509.MarshalPKIXPublicKey(
+			&rsa.PublicKey{
+				N: new(big.Int).SetBytes(nb),
+				E: RSADefaultExponent,
+			},
+		)
+		if err != nil {
+			return "", errors.New(err.Error())
```
Comment (daiLlew) [inline jwks/jwks.go]: Not sure this is required as its not adding any additional context. You can simply return `error`.

File: jwks/jwks.go [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+		}
+	
+		return base64.StdEncoding.EncodeToString(der), nil
+	} else {
+		return "", errors.New(models.JWKSExponentErrorDescription)
+	}
```
Comment (daiLlew) [inline jwks/jwks.go]: Else block is not required here you can simply `return "", errors.New(models.JWKSExponentErrorDescription)` after the if and it achieves the same thing minus the additional indentation/conditional logic.

**Review by daiLlew — COMMENTED** [review]
Just some minor changes required.

**Review by Marianasilvago — APPROVED** [review]
LGTM

**Review by daiLlew — APPROVED** [review]

**Pull request merged**